CMakeLists.txt

```diff
@@ -157,6 +157,12 @@ ADD_SUBDIRECTORY(contrib/utilities)
 
 IF(DEAL_II_HAVE_TESTS_DIRECTORY)
   ADD_SUBDIRECTORY(tests)
+
+  # install tests/tests.h so that it can be accessed in user-code tests
+  CONFIGURE_FILE(
+    ${CMAKE_CURRENT_SOURCE_DIR}/tests/tests.h
+    ${CMAKE_CURRENT_BINARY_DIR}/include/deal.II/tests/tests.h
+    )
 ENDIF()
```

Comment:
I am worried about this conditional: `DEAL_II_HAVE_TESTS_DIRECTORY` is false when the `tests` directory is not present. If you happen to now build and install deal.II without the tests directory you will end up not installing this header file and potentially break downstream projects.

What about we identify the useful and stable part of this header and simply migrate it to `includes`?

Reply:
I can live with that as long as I don't have too see another project that simply copies the `test.h` file because they want to use the deal.II test infrastructure but crucial functions are missing.